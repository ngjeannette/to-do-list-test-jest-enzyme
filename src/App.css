.App {
  text-align: center;
}

.App-header {
  /* background-color: #282c34; */
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

/* checkList */
.checklist {
    overflow: auto;
    max-height: 100px;
}

.total {
  background: #ededed;
}

.input-container {
  display:flex;
  margin: 10px 0;
}