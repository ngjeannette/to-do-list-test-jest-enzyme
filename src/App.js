import './App.css';
import { useState } from 'react';
import List from './List'

function App() {
  const [search, setSearch] = useState('');
  const [list, setList] = useState([]);

  const handleSubmit = () => {
    setList([...list, search]);
    setSearch('')
  }

  const handleInputChange = (e) => {
    setSearch(e.target.value);
  }

  return (
    <div className="App">
      <div className="App-header">
        <div>
            <h1 className="has-text-black is-size-1">To Do Test</h1>
            <div className="input-container">
              <input className="input" type="text" placeholder="Feed the 🐱" onChange={handleInputChange} value={search} />
              <button className="button is-link" id="submit" onClick={handleSubmit}> Submit</button>
            </div>
            <List list={list} />
        </div>
      </div>
    </div>
  );
}

export default App;
