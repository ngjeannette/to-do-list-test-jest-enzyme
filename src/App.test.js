// ============ FULL RENDERING MOUNT instead of shallow =================
import { mount } from 'enzyme';
import App from './App';
import renderer from 'react-test-renderer';
//note: enzyme not compatible with react 17, will have to downgrade to react 16.

describe('To Do Testing', () => {
    let wrapper;
    beforeEach(() => {
        // test app + components, full rendering
        wrapper = mount(<App />)
    })

    // 1) test to render h1 tags UI
    test('render the title', () => {
        expect(wrapper.find('h1').text()).toContain('To Do Test');
    });

    // 2) test to render submit will add onto list
    test('click submit will add onto list', () => {
        wrapper.find('#submit').simulate('click');
        expect(wrapper.find('.panel-block').length).toBe(2);
    })

    // 3) test click of check will check checkbox, and click again will remove checkbox... checking checkbox checks checkbox :D
    test('click of check will toggle checkbox true', () => {
        wrapper.find('#submit').simulate('click');
        wrapper.find('input[type="checkbox"]').simulate("click", { target: { checked: true } });
        wrapper.find('input[type="checkbox"]').simulate("click", { target: { checked: false } });
    })

    // 4) test click of check update complete total + 1
    test('check of checkbox updates complete', () => {
        wrapper.find('#submit').simulate('click');
        wrapper.find('input[type="checkbox"]').simulate("click", { target: { checked: true } });
        wrapper.update();
        expect(parseInt(wrapper.find('#total').text())).toBe(1);
    })
    // 5) test click of check update complete total - 1
    test('uncheck of checkbox updates complete', () => {
        wrapper.find('#submit').simulate('click');
        wrapper.find('input[type="checkbox"]').simulate("click", { target: { checked: true } });
        wrapper.find('input[type="checkbox"]').simulate("click", { target: { checked: false } });
        wrapper.update();
        expect(parseInt(wrapper.find('#total').text())).toBe(0);
    })
    // 6) snapshot
    test('renders snap shot correctly', () => {
        const tree = renderer
            .create(<App />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
})

// to run test: yarn run test