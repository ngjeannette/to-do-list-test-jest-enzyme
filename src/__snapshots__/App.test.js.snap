// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To Do Testing renders snap shot correctly 1`] = `
<div
  className="App"
>
  <div
    className="App-header"
  >
    <div>
      <h1
        className="has-text-black is-size-1"
      >
        To Do Test
      </h1>
      <div
        className="input-container"
      >
        <input
          className="input"
          onChange={[Function]}
          placeholder="Feed the 🐱"
          type="text"
          value=""
        />
        <button
          className="button is-link"
          id="submit"
          onClick={[Function]}
        >
           Submit
        </button>
      </div>
      <nav
        className="panel"
      >
        <p
          className="panel-heading"
        >
          Keeping in Track
        </p>
        <div
          className="checklist"
        />
      </nav>
    </div>
  </div>
</div>
`;
